### PR TITLE
Update hosted (AzDO) and BYOC (dnceng) agents used in our builds

### PR DESCRIPTION
**Pull request by dougbu**
- dotnet/aspnetcore-internal#3540

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -51,10 +51,10 @@ stages:
         pool:
           ${{ if eq(variables['System.TeamProject'], 'public') }}:
             name: NetCorePublic-Pool
-            queue: BuildPool.Windows.10.Amd64.VS2017.Open
+            queue: BuildPool.Server.Amd64.VS2017.Open
```
Comment (dougbu) [inline azure-pipelines.yml]: @MattGal to confirm: The VS 2017 queues should be around for a while?

Comment (MattGal) [reply azure-pipelines.yml]: As long as there are users who need it and a product is not end-of-life, you can expect we'll keep it around.  Generally we only remove support for stuff like this once it's properly dead or everyone's off it.

I would not be terribly surprised however, that a new feature might come along (as happened for 2015->2017 with the SourceLink feature) where you end up strongly encouraged to adopt a new version of VS.

**Comment by dougbu** [comment]
@pranavkm @SteveSandersonMS I'm not sure about the C-related errors in a JavaScript project. What needs to change?
```
/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj(23,5): error : Error running install script for optional dependency: "/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/node_modules/fsevents: Command failed.
##[error]src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj(23,5): error : Error running install script for optional dependency: "/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/node_modules/fsevents: Command failed.
/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj(23,5): error : Error running install script for optional dependency: "/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/node_modules/fsevents: Command failed.
##[error]src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj(23,5): error : Error running install script for optional dependency: "/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/node_modules/fsevents: Command failed.
...
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
    SOLINK_MODULE(target) Release/.node
clang : error : libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated] [/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj]
##[error]clang(0,0): error : libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated]
clang : error : libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated] [/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj]
##[error]clang(0,0): error : libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated]
    CXX(target) Release/obj.target/fse/fsevents.o
YARN : error : include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found] [/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj]
##[error]YARN(0,0): error : include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
YARN : error : include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found] [/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj]
##[error]YARN(0,0): error : include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
  In file included from ../fsevents.cc:6:
```

(I suspect the `fsevents` package is out-of-date but am not sure of the right incantation to upgrade it.)

**Comment by dougbu** [comment]
Hmm, 'master' built successfully on Monday. Either a very recent change in the C++ or JS worlds is affecting us or the hosted `macOS-10.14` agents have **older** tooling than the (deprecated) `macOS-10.13` agents we were using.

**Comment by dougbu** [comment]
@pranavkm @SteveSandersonMS any suggestions on the macOS failures after moving to 10.14?
```
/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/Microsoft.AspNetCore.Components.Electron.JS.csproj(23,5): error : Error running install script for optional dependency: "/Users/runner/runners/2.165.0/work/1/s/src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron.JS/node_modules/fsevents: Command failed.
```

**Comment by dougbu** [comment]
Thanks @pranavkm❕❕

**Pull request merged**